lib/core/scheduler.rb

```diff
@@ -83,51 +80,7 @@ def sync_triggered?(connector_settings)
         return true
       end
 
-      # Don't sync if sync is explicitly disabled
-      scheduling_settings = connector_settings.scheduling_settings
-      unless scheduling_settings.present? && scheduling_settings[:enabled] == true
-        Utility::Logger.debug("#{connector_settings.formatted.capitalize} scheduling is disabled.")
-        return false
-      end
-
-      current_schedule = scheduling_settings[:interval]
-
-      # Don't sync if there is no actual scheduling interval
-      if current_schedule.nil? || current_schedule.empty?
-        Utility::Logger.warn("No sync schedule configured for #{connector_settings.formatted}.")
-        return false
-      end
-
-      current_schedule = begin
-        Utility::Cron.quartz_to_crontab(current_schedule)
-      rescue StandardError => e
-        Utility::ExceptionTracking.log_exception(e, "Unable to convert quartz (#{current_schedule}) to crontab.")
-        return false
-      end
-      cron_parser = Fugit::Cron.parse(current_schedule)
-
-      # Don't sync if the scheduling interval is non-parsable
-      unless cron_parser
-        Utility::Logger.error("Unable to parse sync schedule for #{connector_settings.formatted}: expression #{current_schedule} is not a valid Quartz Cron definition.")
-        return false
-      end
-
-      # We want to sync when sync never actually happened
-      last_synced = connector_settings[:last_synced]
-      if last_synced.nil? || last_synced.empty?
-        Utility::Logger.info("#{connector_settings.formatted.capitalize} has never synced yet, running initial sync.")
-        return true
-      end
-
-      next_trigger_time = cron_parser.next_time(Time.parse(last_synced))
-
-      # Sync if next trigger for the connector is in past
-      if next_trigger_time < Time.now
-        Utility::Logger.info("#{connector_settings.formatted.capitalize} sync is triggered by cron schedule #{current_schedule}.")
-        return true
-      end
-
-      false
+      schedule_triggered?(connector_settings.scheduling_settings, connector_settings.formatted, connector_settings.last_synced)
```

Comment:
What is the fundamental difference between `scheduling_settings` and `custom_scheduling`? I have a feeling that `scheduling_settings` might be just a part of `custom_scheduling` and then we can have something like this: 

```JSON
scheduling_settings: {
  key1: { ...configuration }
  key2: { ...configuration }
}
```

It will help us to avoid having 2 similar abstractions in the protocol and in the source code.

Reply:
Originally this was what I wanted to do (move everything to within `scheduling` and from 8.7 onwards multiple schedules would be supported). It was brought up in the implementation document that it lacks backwards compatibility. So the solution was to have a "main" scheduling (what already exists) and `custom_scheduling` would belong under its own namespace.

The fundamental difference: `custom_scheduling` is a subset of `scheduling` that runs partial jobs that override top-level settings.

There's a conversation still open on the implementation document about this right now: https://docs.google.com/document/d/1F7SCENHYKpG58ZDPZRrcrqctBJhYt8tUOAxMiI_M1sU/edit#

Reply:
I left a comment in the doc. Let's proceed with the current solution but could you please create an issue with the tech debt label?

Reply:
@vidok good idea, I've created the issue here: https://github.com/elastic/enterprise-search-team/issues/3808